dist/ads.js

```diff
@@ -127,32 +135,47 @@ const parseSellersJSON = async (response) => {
             seller_count: 0,
           }
         },
-        passthrough_count: 0
+        passthrough_count: 0,
+        confidential_count: 0
       }
     };
 
     // Clean up file content
     result.seller_count = content.sellers.length;
 
     for (let seller of content.sellers) {
+      // Validating records
+      if (!seller.seller_type && !seller.id) {
+        continue;
+      }
+
+      // Passthrough
+      if (seller.is_passthrough) {
+        result.passthrough_count += 1;
+      }
+
+      // Confidential
+      if (seller.is_confidential) {
+        result.confidential_count += 1;
+      }
+      
+      if (!seller.domain) {
+        continue;
+      }
+
       // Seller records
       let type = seller.seller_type.trim().toLowerCase(),
         domain = seller.domain.trim().toLowerCase();
       if (Object.keys(result.seller_types).includes(type)) {
         result.seller_types[type].domains.add(domain);
         result.seller_types[type].seller_count += 1;
       }
```

Comment:
```suggestion
      if (seller.domain) {
        const domain = seller.domain.trim().toLowerCase();
        result.seller_types[type].domains.add(domain);
        result.seller_types[type].seller_count += 1;
      }
```

Did the check for types above already.